internal/command/group_grant.go

```diff
@@ -0,0 +1,449 @@
+package command
+
+import (
+	"context"
+	"reflect"
+
+	"github.com/zitadel/zitadel/internal/domain"
+	"github.com/zitadel/zitadel/internal/eventstore"
+	"github.com/zitadel/zitadel/internal/repository/groupgrant"
+	"github.com/zitadel/zitadel/internal/repository/org"
+	"github.com/zitadel/zitadel/internal/repository/project"
+	"github.com/zitadel/zitadel/internal/telemetry/tracing"
+	"github.com/zitadel/zitadel/internal/zerrors"
+)
+
+func (c *Commands) AddGroupGrant(ctx context.Context, groupgrant *domain.GroupGrant, resourceOwner string) (_ *domain.GroupGrant, err error) {
+	ctx, span := tracing.NewSpan(ctx)
+	defer func() { span.EndWithError(err) }()
+
+	event, addedGroupGrant, err := c.addGroupGrant(ctx, groupgrant, resourceOwner)
+	if err != nil {
+		return nil, err
+	}
+	pushedEvents, err := c.eventstore.Push(ctx, event)
```

Comment:
In generall, you can use `c.pushAppendAndReduce` if you don't want anything to happend in between the push and append.